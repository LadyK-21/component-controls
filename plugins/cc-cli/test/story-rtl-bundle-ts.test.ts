import path from 'path';
import { runTests } from './run-story-tests';

runTests({
  format: 'ts',
  renderer: 'rtl',
  bundle: path.resolve(__dirname, 'bundle/component-controls.js'),
});
