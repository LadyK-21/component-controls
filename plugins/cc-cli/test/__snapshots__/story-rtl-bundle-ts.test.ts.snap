// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rtl bundle ts 1`] = `
"import * as path from 'path';
const { run, AxeResults } = require('axe-core');
const { reactRunDOM } = require('@component-controls/test-renderers');
import '@component-controls/jest-axe-matcher';
import { loadStore } from '@component-controls/store';
import { render, act } from '@testing-library/react';
import { renderErr } from '@component-controls/test-renderers';

describe('Components/Header', () => {
  const store = loadStore(
    require(path.resolve(__dirname, './bundle/component-controls.js')),
  );
  const renderFn = store.config.renderFn;
  const doc = store.docs['Components/Header'];

  describe('Overview', () => {
    const story = store.stories['components-header--overview'];

    let rendered;
    act(() => {
      rendered = renderFn(story, doc);
    });
    if (!rendered) {
      renderErr();
      return;
    }
    it('snapshot', () => {
      const { asFragment } = render(rendered);
      expect(asFragment()).toMatchSnapshot();
    });
    it('accessibility', async () => {
      const axeResults = await reactRunDOM<AxeResults>(rendered, run);
      expect(axeResults).toHaveNoAxeViolations();
    });
  });
});
"
`;
